pkg/resource/template/resource.go

```diff
@@ -214,11 +227,14 @@ func (t *TemplateResource) sync() error {
 					return rerr
 				}
 				err := ioutil.WriteFile(t.Dest, contents, t.FileMode)
-				// make sure owner and group match the temp file, in case the file was created with WriteFile
-				os.Chown(t.Dest, t.Uid, t.Gid)
 				if err != nil {
 					return err
 				}
+				// make sure owner and group match the temp file, in case the file was created with WriteFile
+				if err := os.Chown(t.Dest, t.Uid, t.Gid); err != nil {
```

Comment:
Seems quite confusing having a bunch of locally scoped errs and the function scoped one defined in the return value